Tools/TransformTool/TransformTool.cs

```diff
@@ -537,12 +539,22 @@ public void DropHeldObjects(Node node)
 			this.ClearSnappingState(rayOrigin);
 		}
 
-		void Translate(Vector3 delta, Transform rayOrigin, bool constrained)
+		void Translate(Vector3 delta, Transform rayOrigin, ConstrainedAxis constraints)
 		{
-			if (constrained)
-				m_TargetPosition += delta;
-			else
-				this.ManipulatorSnap(rayOrigin, Selection.transforms, ref m_TargetPosition, ref m_TargetRotation, delta);
+			switch (constraints)
+			{
+				case ConstrainedAxis.X | ConstrainedAxis.Y:
+				case ConstrainedAxis.Y | ConstrainedAxis.Z:
+				case ConstrainedAxis.X | ConstrainedAxis.Z:
```

Comment:
No. This happens in the handle itself. The reason we are doing this is that we don't actually have a solution for snapping the planar handles yet. My original approach was just to treat them as a snap in one axis, then a snap in the other, but that felt unpredictable. When I discussed this with Timoni, we agreed that since the planar handles already "snap" to a plane, it is not as important that they snap to surfaces.